jetty-http2/http2-common/src/main/java/org/eclipse/jetty/http2/HTTP2Connection.java

```diff
@@ -290,4 +332,143 @@ public InvocationType getInvocationType()
             return InvocationType.EITHER;
         }
     }
+
+    private class ParserListener implements Parser.Listener
+    {
+        private final Parser.Listener listener;
+
+        private ParserListener(Parser.Listener listener)
+        {
+            this.listener = listener;
+        }
+
+        @Override
+        public void onData(DataFrame frame)
+        {
+            NetworkBuffer buffer = producer.buffer;
+            buffer.retain();
```

Comment:
where is the matching release for this retain? No callback no network buffer?  Can you at least document who should do the matching release?